### PR TITLE
GraphQL: Add File & LogInWith

### PR DESCRIPTION
**Pull request by Moumouls**
Currently available on parse-server:master

### CODE REVIEW

File: _includes/graphql/files.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 The GraphQL API supports file upload via [GraphQL Upload](https://github.com/jaydenseric/graphql-upload), to send a `File` through `GraphQL` it's recommended to use the [Apollo Upload Client](https://github.com/jaydenseric/apollo-upload-client).
 
-## Add a File field
+## Add a File field to Class
```
Comment (TomWFox) [inline _includes/graphql/files.md]:
```suggestion
## Add a File field to a Class
```

File: _includes/graphql/files.md [hunk 2]
```diff
@@ -42,9 +42,9 @@ mutation updateGameScoreClass {
 }
 ```
 
-## Create and add File
+## Create File
```
Comment (TomWFox) [inline _includes/graphql/files.md]:
```suggestion
## Create a File
```

File: _includes/graphql/files.md [hunk 3]
```diff
@@ -42,9 +42,9 @@ mutation updateGameScoreClass {
...
 
-Currently the GraphQL API does not support nested mutation for the `File` type, so we need to send the file and then create/update the `GameScore` object with the returned information.
+GraphQL API support nested mutation for the `File` type, so you can send the file along the Parse Object or just upload the file and get returned information.
```
Comment (TomWFox) [inline _includes/graphql/files.md]:
```suggestion
The GraphQL API supports nested mutation for the `File` type, so you can send the file along with the Parse Object or just upload the file and get the returned information.
```

File: _includes/graphql/files.md [hunk 4]
```diff
@@ -126,3 +131,51 @@ mutation createGameScore {
   }
 }
 ```
+
+## Create and add a file
+Lets create a new `GameScore` object and upload the file along the creation.
```
Comment (TomWFox) [inline _includes/graphql/files.md]:
```suggestion
Lets create a new `GameScore` object and upload the file.
```

File: _includes/graphql/users.md [hunk 5]
```diff
@@ -103,6 +104,59 @@ mutation logIn {
 
 Note that, when the user logs in, Parse Server generates a new `sessionToken` for future operations.
 
+## 3rd Party Auth
+
+You can log in a user via a 3rd party auth system (Custom, Facebook, Twitter, etc...) with the `logInWith` mutation.
```
Comment (TomWFox) [inline _includes/graphql/users.md]:
```suggestion
You can log in a user via a [3rd party authentication](https://docs.parseplatform.org/parse-server/guide/#supported-3rd-party-authentications) system (Facebook, Twitter, Apple and many more) with the `logInWith` mutation.
```

File: _includes/graphql/users.md [hunk 6]
```diff
@@ -103,6 +104,59 @@ mutation logIn {
...
+For more information about the use of authData: [AuthData Guide](https://docs.parseplatform.org/parse-server/guide/#supported-3rd-party-authentications)
```
Comment (TomWFox) [inline _includes/graphql/users.md]:
```suggestion
```

File: _includes/graphql/users.md [hunk 7]
```diff
@@ -103,6 +104,59 @@ mutation logIn {
 
 Note that, when the user logs in, Parse Server generates a new `sessionToken` for future operations.
 
+## 3rd Party Auth
```
Comment (TomWFox) [inline _includes/graphql/users.md]:
```suggestion
## 3rd Party Authentication
```

**Review by TomWFox — CHANGES_REQUESTED** [review]

**Review by TomWFox — APPROVED** [review]
I’ve applied my changes and I’m happy. @davimacedo do you want to take a look?

I can resolve the bundle.js conflict when merging.

**Review by davimacedo — APPROVED** [review]
It looks good to me. LGTM!

**Pull request merged**